packages/fast-usdc/src/exos/advancer.js

```diff
@@ -9,116 +11,219 @@ import { addressTools } from '../utils/address.js';
-      },
+      advancer: M.interface('AdvancerI', {
+        handleTransactionEvent: M.callWhen(CctpTxEvidenceShape).returns(
+          M.or(M.undefined(), VowShape),
+        ),
+      }),
+      ...WatcherHandlersShape,
     },
-  );
-
-  return zone.exoClass(
-    'Fast USDC Advancer',
-    M.interface('AdvancerI', {
-      handleTransactionEvent: M.call(CctpTxEvidenceShape).returns(VowShape),
-    }),
     /**
      * @param {{
+     *   assetManagerFacet: AssetManagerFacet;
      *   localDenom: Denom;
-     *   poolAccount: HostInterface<OrchestrationAccount<{ chainId: 'agoric' }>>;
+     *   poolAccount: ERef<HostInterface<OrchestrationAccount<{chainId: 'agoric';}>>>
+     *   usdcBrand: Brand<'nat'>;
      * }} config
      */
     config => harden(config),
     {
-      /** @param {CctpTxEvidence} evidence */
-      handleTransactionEvent(evidence) {
-        // TODO EventFeed will perform input validation checks.
-        const { recipientAddress } = evidence.aux;
-        const { EUD } = addressTools.getQueryParams(recipientAddress).params;
-        if (!EUD) {
-          statusManager.observe(evidence);
-          throw makeError(
-            `recipientAddress does not contain EUD param: ${q(recipientAddress)}`,
-          );
-        }
+      advancer: {
+        /**
+         * Returns a Promise for a Vow in the happy path, or undefined
+         * when conditions are not met. Aims to perform a status update for
+         * every observed transaction.
+         *
+         * We do not expect any callers to depend on the settlement of
+         * `handleTransactionEvent` - errors caught are communicated to the
+         * `StatusManager` - so we don't need to concern ourselves with
+         * preserving the vow chain for callers.
+         *
+         * @param {CctpTxEvidence} evidence
+         */
+        async handleTransactionEvent(evidence) {
+          await null;
+          try {
+            // TODO poolAccount might be a vow we need to unwrap
+            const { assetManagerFacet, poolAccount, usdcBrand } = this.state;
+            const { recipientAddress } = evidence.aux;
+            const { EUD } =
+              addressTools.getQueryParams(recipientAddress).params;
+            if (!EUD) {
+              throw makeError(
+                `recipientAddress does not contain EUD param: ${q(recipientAddress)}`,
+              );
+            }
 
-        // TODO #10391 this can throw, and should make a status update in the catch
-        const destination = chainHub.makeChainAddress(EUD);
+            // this will throw if the bech32 prefix is not found, but is handled by the catch
+            const destination = chainHub.makeChainAddress(EUD);
 
-        /** @type {DenomAmount} */
-        const requestedAmount = harden({
-          denom: this.state.localDenom,
-          value: BigInt(evidence.tx.amount),
-        });
+            const requestedValue = BigInt(evidence.tx.amount);
```

Comment:
the type says it's already bigint. If you don't trust that property then don't trust any of the object.

Reply:
Will remove the cast